sklearn/preprocessing/imputation.py

```diff
@@ -189,6 +202,15 @@ def _sparse_fit(self, X, strategy, missing_values, axis):
         else:
             X = X.tocsc()
 
+        if axis == 0:
+            mask = _get_mask(X.data, self.missing_values)
+            mask_matrix = X.__class__((mask, X.indices.copy(),
+                                       X.indptr.copy()), shape=X.shape,
+                                      dtype=X.dtype)
+            feat_with_missing = mask_matrix.sum(axis=0).A.nonzero()[1]
+            # ravel since nonzero returns 2d matrices for sparse in scipy 0.11
+            feat_with_missing = feat_with_missing.ravel()
```

Comment:
It should rather be `feat_with_missing.A.ravel()` to convert `feat_with_missing` from matrix (not sparse) to array, then flatten.